<?xml version="1.0"?>
<launch>
    <!-- CONTROLLER PARAMETERS -->
    <arg name="robot_frame" default="base_footprint"/>
    <arg name="parent_frame" default="odom"/>
    <arg name="figure8_duration" default="50.0"/>
    <arg name="robot_idx" default="1"/>

    <param name="figure8_duration" value="$(arg figure8_duration)"/>

<!--    <node name="controller" pkg="create2_controller" type="trajectory_controller_v2" output="screen" required="true">-->
<!--        <param name="frame" value="$(arg robot_frame)"/>-->
<!--        <param name="parentFrame" value="$(arg parent_frame)"/>-->
<!--        <param name="Kx" value="5.0" />-->
<!--        <param name="Ky" value="5.0" />-->
<!--        <param name="Ktheta" value="40.0"/>-->
<!--    </node>-->

    <node name="figure8" pkg="rlss_ros" type="figure8" output="screen" required="true">
    </node>

    <arg name="model" default="burger"/>
    <arg name="x_pos" default="0.0"/>
    <arg name="y_pos" default="0.0"/>
    <arg name="z_pos" default="0.0"/>

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find turtlebot3_gazebo)/worlds/empty.world"/>
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="true"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
    </include>

    <node pkg="rqt_plot" type="rqt_plot" name="rqt_plot"
        args="/desired_state/position/x /desired_state/position/y /odom/pose/pose/position/x /odom/pose/pose/position/y">

    </node>

    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
    <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model turtlebot3_$(arg model)_$(arg robot_idx) -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -param robot_description" />

</launch>